### PR TITLE
use correct version copmparison on NSIS updater: fix update from rc

### PR DESCRIPTION
**Pull request by mgallien**
Fix https://github.com/nextcloud/client_updater_server/issues/96

Signed-off-by: Matthieu Gallien <matthieu.gallien@nextcloud.com>

### CODE REVIEW

File: src/gui/updater/ocupdater.cpp [hunk 1]
```diff
@@ -361,28 +362,32 @@ void NSISUpdater::versionInfoArrived(const UpdateInfo &info)
...
         setDownloadState(UpToDate);
-    } else if (infoVersion <= currVersion
-               || infoVersion <= seenVersion) {
-        qCInfo(lcUpdater) << "Client is on latest version!";
-        setDownloadState(UpToDate);
     } else {
-        QString url = info.downloadUrl();
-        if (url.isEmpty()) {
-            showNoUrlDialog(info);
+        qint64 currentVer = Helper::currentVersionToInt();
```
Comment (allexzander) [inline src/gui/updater/ocupdater.cpp:366]: `const auto`

File: src/gui/updater/ocupdater.cpp [hunk 2]
```diff
@@ -361,28 +362,32 @@ void NSISUpdater::versionInfoArrived(const UpdateInfo &info)
...
+        qint64 remoteVer = Helper::stringVersionToInt(info.version());
```
Comment (allexzander) [inline src/gui/updater/ocupdater.cpp:367]: same

File: src/gui/updater/ocupdater.cpp [hunk 3]
```diff
@@ -361,28 +362,32 @@ void NSISUpdater::versionInfoArrived(const UpdateInfo &info)
...
+
+        if (info.version().isEmpty() || currentVer >= remoteVer) {
+            qCInfo(lcUpdater) << "Client is on latest version!";
+            setDownloadState(UpToDate);
         } else {
-            _targetFile = cfg.configPath() + url.mid(url.lastIndexOf('/')+1);
-            if (QFile(_targetFile).exists()) {
-                setDownloadState(DownloadComplete);
+            QString url = info.downloadUrl();
```
Comment (allexzander) [inline src/gui/updater/ocupdater.cpp:373]: `const auto`

File: src/gui/updater/ocupdater.cpp [hunk 4]
```diff
@@ -259,6 +259,7 @@ bool OCUpdater::updateSucceeded() const
 
 void OCUpdater::slotVersionInfoArrived()
 {
+    qCInfo(lcUpdater()) << "received a reply";
```
Comment (allexzander) [inline src/gui/updater/ocupdater.cpp:262]: `qCInfo(lcUpdater())` -> `qCInfo(lcUpdater)`

**Review by allexzander — CHANGES_REQUESTED** [review]
@matthieugallien 
Besides some nitpicks, I've got question:

What about the feature of skipping a certain version? I see that now `seenVersion` is no longer in use, is this expected? Also, if this is the case, the entire code for `seenVersion` needs to be removed as well to avoid confusion?

**Comment by mgallien** [comment]
> @matthieugallien Besides some nitpicks, I've got question:
> 
> What about the feature of skipping a certain version? I see that now `seenVersion` is no longer in use, is this expected? Also, if this is the case, the entire code for `seenVersion` needs to be removed as well to avoid confusion?

I am not convinced we should fix the current interface to tell users about the skipped version. I would prefer to remove it altogether as we should aim to produce new versions that are not broken and skipping an update should never be needed.

**Review by allexzander — APPROVED** [review]

**Pull request merged**